Analysis/Tutorials/src/partitions.cxx

```diff
@@ -0,0 +1,109 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+
+namespace o2::aod
+{
+namespace etaphi
+{
+DECLARE_SOA_COLUMN(Eta, eta2, float);
+DECLARE_SOA_COLUMN(Phi, phi2, float);
+} // namespace etaphi
+DECLARE_SOA_TABLE(EtaPhi, "AOD", "ETAPHI",
+                  etaphi::Eta, etaphi::Phi);
+} // namespace o2::aod
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+// This is a very simple example showing how to iterate over tracks
+// and create a new collection for them.
+// FIXME: this should really inherit from AnalysisTask but
+//        we need GCC 7.4+ for that
+struct TTask {
+  Produces<aod::EtaPhi> etaphi;
+
+  // FIXME: For some reason filtering with Charge does not work??
+  //Partition<aod::Tracks> negativeTracksP = aod::track::Charge < 0;
+  //Partition<aod::Tracks> positiveTracksP = aod::track::Charge > 0;
+  Partition<aod::Tracks> negativeTracksP = aod::track::pt2 < 1.0f;
+  Partition<aod::Tracks> positiveTracksP = aod::track::pt2 > 1.0f;
```

Comment:
Thanks for the work done. A couple of questions:

* Can we actually create the partitions inside the process function?
* Can we get the negated partition by simply saying `!`?

Reply:
Creating inside process() is possible, just one needs to write also: `negativeTracksP.setTable(tracks)` afterwards.
Or, I can adjust the code to what is proposed in ANALYSIS.md, i.e. `auto& negativeTracks =  negativeTracksP(tracks)`, that would replace also lines 48-49

I don't think negation is possible, this would need to return a boolean

Reply:
For creating inside process we also need to define some interactions with filters - currently partitioning will happen on a full table always. But if user requests a Filtered tracks, they would also expect partitioning happen on top of that. For now, I would skip the ability to create partitions in process(). The negation is relatively easy to implement, though, but it should be an operator! on Filtered.

Reply:
Well, even when Partition is created before process(), it does not compile together with Filters (trying to filter the Filtered).

Reply:
This you can avoid if you use T::table_t as a type, not T directly, which should be always soa::Table.